cfme/physical/physical_server.py

```diff
@@ -68,58 +69,90 @@ def load_details(self, refresh=False):
             view.browser.refresh()
             view.flush_widget_cache()
 
-    def execute_button(self, button_group, button, handle_alert=False):
+    def execute_button(self, button, option, handle_alert=False):
         view = navigate_to(self, "Details")
-        view.toolbar.custom_button(button_group).item_select(button, handle_alert=handle_alert)
+        view.toolbar.custom_button(button).item_select(option, handle_alert=handle_alert)
+        return view
 
-    def power_on(self):
-        view = navigate_to(self, "Details")
-        view.toolbar.power.item_select("Power On", handle_alert=True)
+    def execute_action_button(self, button, option, **kwargs):
+        target = kwargs.get('target', None)
+        provider = kwargs.get('provider', None)
+        desired_state = kwargs.get('desired_state', None)
 
-    def power_off(self):
-        view = navigate_to(self, "Details")
-        view.toolbar.power.item_select("Power Off", handle_alert=True)
+        view = self.execute_button(button, option, handle_alert=True)
+
+        if desired_state:
+            self.wait_for_state_change(desired_state, target, provider, view)
+        else:
+            wait_for(
+                lambda: view.flash.is_displayed,
+                message="Wait for the handle alert to appear...",
+                num_sec=5,
+                delay=2
+            )
+        return view.flash.messages
+
+    def power_on(self, **kwargs):
+        return self.execute_action_button('Power', 'Power On', **kwargs)
+
+    def power_off(self, **kwargs):
+        return self.execute_action_button('Power', 'Power Off', **kwargs)
+
+    def power_off_immediately(self, **kwargs):
+        return self.execute_action_button('Power', 'Power Off Immediately', **kwargs)
+
+    def restart(self, **kwargs):
+        return self.execute_action_button('Power', 'Restart', **kwargs)
+
+    def restart_immediately(self, **kwargs):
+        return self.execute_action_button('Power', 'Restart Immediately', **kwargs)
+
+    def refresh(self, provider, handle_alert=False):
+        last_refresh = provider.last_refresh_date()
+        self.execute_button('Configuration', 'Refresh Relationships and Power States', handle_alert)
+        wait_for(
+            lambda: last_refresh != provider.last_refresh_date(),
+            message="Wait for the server to be refreshed...",
+            num_sec=300,
+            delay=5
+        )
+
+    def lifecycle_provision(self):
+        return navigate_to(self, "Provision")
```

Comment:
**REQUIRED**
This method doesn't make sense. In the test just use `view = navigate_to(self, "Provision")`

Reply:
I change the body of this method, now make sense for you? It's the same approach than the other methods, as `power_on'.